pkg/operator/operatorclient/operatorclient.go

```diff
@@ -19,10 +29,163 @@ import (
 type OperatorClient struct {
 	Informers operatorclientinformers.SharedInformerFactory
 	Client    operatorconfigclient.CertManagersGetter
+	clock     clock.PassiveClock
 }
 
 var _ v1helpers.OperatorClient = &OperatorClient{}
 
+func (c OperatorClient) ApplyOperatorSpec(ctx context.Context, fieldManager string, applyConfiguration *applyoperatorv1.OperatorSpecApplyConfiguration) (err error) {
+	if applyConfiguration == nil {
+		return fmt.Errorf("applyConfiguration must have a value")
+	}
+
+	desired := applyconfig.CertManager("cluster")
+	desired.Spec.OperatorLogLevel = applyConfiguration.OperatorLogLevel
+	desired.Spec.LogLevel = applyConfiguration.LogLevel
+	desired.Spec.ManagementState = applyConfiguration.ManagementState
+	desired.Spec.UnsupportedConfigOverrides = applyConfiguration.UnsupportedConfigOverrides
+	desired.Spec.ObservedConfig = applyConfiguration.ObservedConfig
+	instance, err := c.Client.CertManagers().Get(ctx, "cluster", metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+	// do nothing and proceed with the apply
+	case err != nil:
+		return fmt.Errorf("unable to get operator configuration: %w", err)
+	default:
+		original := &applyconfig.CertManagerApplyConfiguration{}
+		//err := managedfields.ExtractInto(instance, internal.Parser().Type("com.github.openshift.cert-manager-operator.api.v1.CertManager"), fieldManager, original, "")
+		//if err != nil {
+		//	return nil
+		//}
+		original.WithName(instance.Name)
+
+		original.WithKind(instance.Kind)
+		original.WithAPIVersion(instance.APIVersion)
+
+		if equality.Semantic.DeepEqual(original, desired) {
+			return nil
+		}
+	}
+
+	_, err = c.Client.CertManagers().Apply(ctx, desired, metav1.ApplyOptions{
+		Force:        true,
+		FieldManager: fieldManager,
+	})
+	if err != nil {
+		return fmt.Errorf("unable to Apply for operator using fieldManager %q: %w", fieldManager, err)
+	}
+
+	return nil
+}
+
+func (c OperatorClient) ApplyOperatorStatus(ctx context.Context, fieldManager string, applyConfiguration *applyoperatorv1.OperatorStatusApplyConfiguration) (err error) {
```

Comment:
Since we're adding custom (non-auto generated) code for this ApplyOperatorStatus which is required required in the deplotment_controller (and possibly other controllers) vendored from library-go, we need to add unit tests here to ensure this is never broken.

For reference, I added a few unit test cases to exactly test such scenarios. Today, the operator only adds/updates into `status.OperatorStatus.Conditions` and `status.OperatorStatus.Generations`.
https://github.com/openshift/cert-manager-operator/pull/266/commits/f1863ac5e0b37a5adc9b9c467a40faecd2e2e00a

Reply:
The fake client unfortunately does not support client.Apply funcs yet (ref: https://github.com/kubernetes-sigs/controller-runtime/issues/2341) which is why the results of testing the implementation against it are not accurate. We can follow-up with [Envtest based integration test](https://book.kubebuilder.io/reference/envtest) instead of unit tests based on fake client, but that'll require some work to add a new CI job where integration tests are run separately.